ccm_web/client/src/pages/FormatThirdPartyGradebook.tsx

```diff
@@ -0,0 +1,446 @@
+    return setRecords(validationResult.validData)
+  }
+
+  const parseFile = (file: File): void => {
+    csvParser.parseCSV(
+      file,
+      handleSchemaValidation,
+      message => setSchemaInvalidations([{ message, type: InvalidationType.Error }])
+    )
+  }
+
+  useEffect(() => {
+    if (file !== undefined) {
+      parseFile(file)
+    }
+  }, [file])
+
+  useEffect(() => {
+    if (studentLoginIds !== undefined && records !== undefined) {
+      const processor = new GradebookProcessor(studentLoginIds)
+      const result = processor.process(records)
+      if (result.valid) {
+        setProcessedRecords(result.processedRecords)
+        setAssignmentHeader(result.assignmentHeader)
+      }
+      setGradebookInvalidations(result.invalidations)
+      if (result.invalidations.length === 0) {
+        setActiveStep(FormatGradebookStep.Review)
+      }
+    }
+  }, [records])
+
+  const handleResetSelect = (): void => {
+    setSections(undefined)
+    clearGetSectionsError()
+    setSelectedSections(undefined)
+    setStudentLoginIds(undefined)
+    clearGetStudentsError()
+  }
+
+  const handleResetUpload = (): void => {
+    setFile(undefined)
+    setRecords(undefined)
+    setSchemaInvalidations(undefined)
+    setProcessedRecords(undefined)
+    setGradebookInvalidations(undefined)
+  }
+
+  const handleFullReset = (): void => {
+    handleResetSelect()
+    handleResetUpload()
+    setActiveStep(FormatGradebookStep.Select)
+  }
+
+  const handleBack = (): void => setActiveStep((prevStep) => {
+    if (prevStep > FormatGradebookStep.Select) return prevStep - 1
+    return prevStep
+  })
+
+  const backButton = <Button className={classes.backButton} onClick={handleBack}>Back</Button>
+
+  const renderSchemaInvalidations = (invalidations: SchemaInvalidation[]): JSX.Element => {
+    const messages = invalidations.map(
+      (invalidation, i) => <Typography key={i}>{invalidation.message}</Typography>
+    )
+    return <ErrorAlert messages={messages} tryAgain={handleResetUpload} />
+  }
+
+  const renderGradebookInvalidations = (invalidations: GradebookInvalidation[]): JSX.Element => {
+    const errors = invalidations.filter(i => i.type === InvalidationType.Error)
+    const warnings = invalidations.filter(i => i.type === InvalidationType.Warning)
+
+    if (errors.length > 0) {
+      return (
+        <ErrorAlert
+          messages={errors.map((e, i) => <Typography key={i}>{e.message}</Typography>)}
+          tryAgain={handleResetUpload}
+        />
+      )
+    }
+    if (warnings.length > 0) {
+      return (
+        <WarningAlert
+          messages={warnings.map((w, i) => <Typography key={i}>{w.message}</Typography>)}
+          cancel={handleResetUpload}
+          cont={() => setActiveStep(FormatGradebookStep.Review)}
+        />
+      )
+    }
+    return <ErrorAlert tryAgain={handleResetUpload} />
+  }
+
+  const renderSelect = (): JSX.Element => {
```

Comment:
Do you feel the variable name should reflect what it is doing? since it is primarily displaying error message and resetting the default. I got confused by the name and action it is doing atleast from my stand point of view

Reply:
Isn't it doing more than that though? If there is no error, then it shows the select component. I could rename it renderSelectStep? I was using some of the naming patterns set up by Chris.

Reply:
Ok, No need to make any changes to keep the naming consistency across features.